packages/loader/container-loader/src/containerContext.ts

```diff
@@ -302,10 +302,11 @@ export class ContainerContext implements IContainerContext {
         this._runtime = await runtimeFactory.instantiateRuntime(this);
     }
 
+    public notifyAttaching() {
+        this._runtime?.setAttachState?.(AttachState.Attaching);
```

Comment:
Consider adding a // back-compat comment here suggesting when to make it non-conditional.
Also I think first ? should be removed - there has to be a runtime by that point, right?

Reply:
Also looking below, it sounds like setAttachState() was in place for a while, so maybe it's time to use it unconditionally?